rfcs/text/0012-orchestrator-field-set.md

```diff
@@ -33,15 +98,98 @@ Stage 3: Add or update all remaining field definitions. The list should now be e
--->
+The `orchestrator` field set will be used to capture typical concepts employed
+by container orchestrators to manage resources. The key intent of this is to create
+a consistent method by which audit logs from container orchestrators can
+be compared. For example, this would allow the creation of open source detection
+rulesets for suspicious Kubernetes events based on audit logs, which can be easily
+transferred from one cluster to another without depending on the specifics of
+parsing implementations.
+
+This might also have use in performance and monitoring tooling which exists around
+container orchestrators, allowing for the definition of shareable dashboards and
+alert definitions.
 
 ## Source data
 
-<!--
-Stage 1: Provide a high-level description of example sources of data. This does not yet need to be a concrete example of a source document, but instead can simply describe a potential source (e.g. nginx access log). This will ultimately be fleshed out to include literal source examples in a future stage. The goal here is to identify practical sources for these fields in the real world. ~1-3 sentences or unordered list.
--->
+Examples of source data include:
+
+- [Kubernetes audit logs][1]
+- [Kubernetes node logs][2]
+- [HashiCorp Nomad audit logs][3]
+- [Falco alert logs][4]
+
+### Kubernetes audit log
+
+```json
+{
+  "_index": "filebeat-7.7.0-2020.12.31-000001",
+  "_type": "_doc",
+  "_id": "KbmPuXYBaTdcl42uyGfl",
+  "_version": 1,
+  "_score": null,
+  "_source": {
+    "@timestamp": "2020-12-31T16:09:35.735Z",
+    "log": {
+      "offset": 7248566,
+      "file": {
+        "path": "/tmp/host-logs/kube-apiserver-audit.log"
+      }
+    },
+    "cloud.provider": "gcp",
+    "event.action": "create",
+    "orchestrator.cluster": {
+        "name": "test-dev",
+    }
+    "orchestrator.type": "Kubernetes",
+    "orchestrator.subresource": "attach",
+    "orchestrator.resource.type": "pods",
+    "orchestrator.namespace": "default",
+    "orchestrator.resource.name": "test",
+    "orchestrator.api_version": "v1",
+    "user.name": "system:serviceaccount:test"
+  }
+}
+```
+
+### Hashicorp Nomad audit log
+
+```json
+{
+  "created_at": "2020-03-24T13:09:35.704224536-04:00",
+  "event_type": "audit",
+  "payload": {
+    "id": "8b826146-b264-af15-6526-29cb905145aa",
+    "stage": "OperationComplete",
+    "type": "audit",
+    "timestamp": "2020-03-24T13:09:35.703865005-04:00",
+    "version": 1,
+    "auth": {
+      "accessor_id": "a162f017-bcf7-900c-e22a-a2a8cbbcef53",
+      "name": "Bootstrap Token",
+      "global": true,
+      "create_time": "2020-03-24T17:08:35.086591881Z"
+    },
+    "request": {
+      "id": "02f0ac35-c7e8-0871-5a58-ee9dbc0a70ea",
+      "event.action": "GET",
+      "orchestrator.api_version": "v1",
+      "orchestrator.resource.type": "nodes",
+      "orchestrator.namespace": "default"
```

Comment:
Why are these `orchestrator` fields under `request`? Shouldn't they be at the root level? I would also expect `orchestrator.type: nomad`.